platform/security/platform-security-core-api/src/main/java/ddf/security/common/audit/SecurityLogger.java

```diff
@@ -76,30 +80,52 @@ private static void requestIpAndPortAndUserMessage(
   private static void requestIpAndPortAndUserMessage(
       Subject subject, Message message, StringBuilder messageBuilder) {
     String user = getUser(subject);
-    if (message != null) {
+    messageBuilder.append(SUBJECT).append(user);
+    appendConditionalAttributes(subject, messageBuilder);
+
+    if (message == null) {
+      messageBuilder.append(" ");
+    } else {
       HttpServletRequest servletRequest =
           (HttpServletRequest) message.get(AbstractHTTPDestination.HTTP_REQUEST);
       // pull out the ip and port of the incoming connection so we know
       // who is trying to get access
       if (servletRequest != null) {
         messageBuilder
-            .append(SUBJECT)
-            .append(user)
             .append(" Request IP: ")
             .append(servletRequest.getRemoteAddr())
             .append(", Port: ")
             .append(servletRequest.getRemotePort())
             .append(" ");
       } else if (MessageUtils.isOutbound(message)) {
         messageBuilder
-            .append(SUBJECT)
-            .append(user)
             .append(" Outbound endpoint: ")
             .append(message.get(Message.ENDPOINT_ADDRESS))
             .append(" ");
       }
-    } else {
-      messageBuilder.append(SUBJECT).append(user).append(" ");
+    }
+  }
+
+  /**
+   * Appends any additional attributes as defined in the comma-delimited system property {@link
+   * #EXTRA_ATTRIBUTES_PROP}.
+   *
+   * @param subject the subject of the logging request
+   * @param messageBuilder buffer to which to append attribute text, if any
+   */
+  private static void appendConditionalAttributes(Subject subject, StringBuilder messageBuilder) {
+    String attributes = System.getProperty(EXTRA_ATTRIBUTES_PROP);
+    if (subject == null) {
+      subject = ThreadContext.getSubject();
+    }
+    if (attributes != null) {
+      List<String> attributeList = Arrays.asList(attributes.split(","));
+      for (String attribute : attributeList) {
+        List<String> attributeValueList = SubjectUtils.getAttribute(subject, attribute);
+        if (attributeValueList != null && !attributeValueList.isEmpty()) {
+          messageBuilder.append(" " + attribute + " : " + attributeValueList.get(0));
```

Comment:
❓ Can `attributeValueList` contain more than one value?

Reply:
Good question. From what I've seen, no, but I can't anticipate what the user might need and there could potentially be more than one value. Fixing now.